kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/KeyValueHeaderRecordWriterProvider.java

```diff
@@ -70,9 +70,8 @@ public RecordWriter getRecordWriter(S3SinkConnectorConfig conf, String filename)
     // Each provider will add its own extension. The filename comes in with the value file format,
     // e.g. filename.avro, but when the format class is different for the key or the headers the
     // extension needs to be removed.
-    int extensionIndex = filename.indexOf(valueProvider.getExtension());
-    String strippedFilename = extensionIndex > -1
-        ? filename.substring(0, extensionIndex)
+    String strippedFilename = filename.endsWith(valueProvider.getExtension())
```

Comment:
should we extract this to a protected function and unit test this?  

I like that we have added IT but the unit test would be faster and we can test more combinations whereas IT would do end to end test.